bindings/azure/cosmosdb/cosmosdb.go

```diff
@@ -85,30 +88,24 @@ func (c *CosmosDB) Init(metadata bindings.Metadata) error {
 		}
 	}
 
-	dbClient, err := client.NewDatabase(m.Database)
-	if err != nil {
-		return err
-	}
 	// Create a container client
-	dbContainer, err := dbClient.NewContainer(m.Collection)
+	dbContainer, err := client.NewContainer(m.Database, m.Collection)
 	if err != nil {
 		return err
 	}
 	if dbContainer.ID() == "" {
-		return backoff.Permanent(
-			fmt.Errorf("container is non-existent"),
-		)
+		return fmt.Errorf("container is non-existent")
 	}
 	c.client = dbContainer
-	containerRead, err := c.client.Read(context.Background(), nil)
+	ctx, cancel := context.WithTimeout(context.Background(), timeoutValue*time.Second)
+	containerRead, err := c.client.Read(ctx, nil)
+	cancel()
 	if err != nil {
 		return err
 	}
 
-	if containerRead.ContainerProperties.ResourceID == "" {
-		return backoff.Permanent(
-			fmt.Errorf("attempted to read a non-existent container"),
-		)
+	if containerRead.RawResponse.StatusCode == http.StatusNotFound {
```

Comment:
On my previous comment what I meant was that if the container does not exist, you won't hit this code:

```
containerRead, err := c.client.Read(ctx, nil)
	cancel()
	if err != nil {
		return err
	}
```

A 404 is returned as `err`.

You can return the err to the caller like you currently.

Reply:
Lines 101-108 can go away and just become `return err`

So you'd have

```go
_, err := c.client.Read(ctx, nil)
return err
```